app/models/mixins/retirement_mixin.rb

```diff
@@ -178,6 +181,7 @@ def retired_event_name
   end
 
   def raise_retirement_event(event_name, requester = nil)
+    requester ||= User.current_user.try(:userid)
```

Comment:
The `raise_retirement_event` method gets call from two places: `retirement_check` and `retire_now`.

The `retirement_check` method is called from the scheduler and I do not believe it gets called anywhere else, so I would expect `User.current_user` to be nil.

The `retire_now` method can get called directly by a user and on line 128 sets `update_attributes(:retirement_requester => requester)`.   With your current code change the `User.current_user` would be passed to the event, but not set for the `retirement_requester` on the object, so this is only a partial fix.

Moving `requester ||= User.current_user.try(:userid)` into `retire_now` before the `update_attributes` call would resolve this.

Reply:
```retirement_check``` gets called when running ```Set Retirement Date```.

When ```retire_now``` gets called from UI or API, ```requester``` is already set with this change.